engine/src/test/java/org/hibernate/validator/test/predefinedscope/PredefinedScopeConstraintDefinitionContributorTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.predefinedscope;
+
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertThat;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.violationOf;
+import static org.hibernate.validator.testutils.ValidatorUtil.getValidator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+import org.hibernate.validator.PredefinedScopeHibernateValidator;
+import org.hibernate.validator.test.constraintvalidator.MustMatch;
+import org.hibernate.validator.test.constraintvalidator.MustNotMatch;
+import org.hibernate.validator.testutil.TestForIssue;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test related to constraint validator discovery in the predefined scope case.
+ *
+ * @author Hardy Ferentschik
+ * @author Guillaume Smet
+ */
+@TestForIssue(jiraKey = "HV-1721")
+public class PredefinedScopeConstraintDefinitionContributorTest {
```

Comment:
Maybe we should do this test the same way we have `AbstractConstructorValidationTest` and 2 impl for annotations and xml. In this case it'll be predefined and regular impls.

Reply:
Yeah, the thing is I don't like to change preexisting tests but if we end up having a lot more, I agree we should do something about it.

In general, most of the tests should be tested for both but that would require changing so many things and I don't think we have the bandwidth.

My estimated tolerance is of 3 copied tests :).